packages/react-router-dom/__tests__/data-browser-router-test.tsx

```diff
@@ -1665,14 +1665,12 @@ function testDomRouter(
       let router = createTestRouter(
         createRoutesFromElements(
           <Route element={<Layout />}>
-            <Route index loader={() => "index"} element={<h1>index</h1>} />
```

Comment:
The unit test changes in this file can be ignored - before I added the future flag (and I started treating partial hydration data as non-initialized), I found a bunch of tests that inadvertently forgot or provide partial loader data so cleaned them up